### PR TITLE
Extract runner selection logic

### PR DESCRIPTION
**Pull request by alanmcruickshank**
As suggested in #1101 and to support #1088, this extracts runner selection logic into the `runner` module.

### CODE REVIEW

File: src/sqlfluff/core/linter/runner.py [hunk 1]
```diff
@@ -211,3 +204,28 @@ def __init__(self, ee):
...
+
+
+def get_runner(
+    linter,
+    config,
+    parallel: int,
+    allow_process_parallelism: bool = True,
+) -> BaseRunner:
+    """Generate a runner instance based on parallel and sytem configuration."""
+    if parallel > 1 and sys.version_info > (3, 7):
```
Comment (barrywhart) [inline src/sqlfluff/core/linter/runner.py:216]: I noted this on the other PR as well -- I think the version check should be `sys.version_info > (3, 7)`. (Minor bug.)

**Review by barrywhart — APPROVED** [review]
One minor pre-existing bug I noticed in the code that's being moved. Would be cool to fix but not vital.

File: src/sqlfluff/core/linter/runner.py (hunk 1 above)
Comment (alanmcruickshank) [reply src/sqlfluff/core/linter/runner.py:216]: @barrywhart I'm confused. The check is _already_ `(3, 7)`, did you mean it should be something else? I'd also love to add a comment saying _why_ the earlier versions aren't supported - do you know what the issue is with them?

Comment (alanmcruickshank) [reply src/sqlfluff/core/linter/runner.py:216]: Scratch that.... saw your other comment. Will fix it there.

**Pull request merged**